File: jwt/jwt/src/main/java/com/example/jwt/entity/GamePlay.java
package com.example.jwt.entity;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Getter
@Setter
@AllArgsConstructor
@Builder
@Table(name = "game_play")
public class GamePlay {

    @Id
    private String userUID;

    // 게임별 플레이 횟수
    @Column(nullable = false)
    private int wordGame;

    @Column(nullable = false)
    private int findWrongGame;

    @Column(nullable = false)
    private int danceGame;

    @Column(nullable = false)
    private int paintGame;

    @Column(nullable = false)
    private int makeBookGame;

    public GamePlay() {
        this.wordGame = 0;
        this.findWrongGame = 0;
        this.danceGame = 0;
        this.paintGame = 0;
        this.makeBookGame = 0;
    }
}

File: playschool/lib/src/authentication/model/GamePlay.dart
class GamePlay {
  final String userUID;
  final int wordGame;
  final int findWrongGame;
  final int danceGame;
  final int paintGame;
  final int makeBookGame;

  const GamePlay({
    required this.userUID,
    required this.wordGame,
    required this.findWrongGame,
    required this.danceGame,
    required this.paintGame,
    required this.makeBookGame,
  });

  factory GamePlay.fromJson(Map<String, dynamic> json) {
    return GamePlay(
      userUID: json["userUID"],
      wordGame: json["wordGame"],
      findWrongGame: json["findWrongGame"],
      danceGame: json["danceGame"],
      paintGame: json["paintGame"],
      makeBookGame: json["makeBookGame"],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "userUID" : userUID,
      "wordGame" : wordGame,
      "findWrongGame" : findWrongGame,
      "danceGame" : danceGame,
      "paintGame" : paintGame,
      "makeBookGame" : makeBookGame
    };
  }

  GamePlay copyWith({
    String? userUID,
    int? wordGame,
    int? findWrongGame,
    int? danceGame,
    int? paintGame,
    int? makeBookGame,
  }) {
    return GamePlay(
      userUID: userUID ?? this.userUID,
      wordGame: wordGame ?? this.wordGame,
      findWrongGame: findWrongGame ?? this.findWrongGame,
      danceGame: danceGame ?? this.danceGame,
      paintGame: paintGame ?? this.paintGame,
      makeBookGame: makeBookGame ?? this.makeBookGame,
    );
  }
}